New frontend HTML/CSS theme

mizdebsk opened the pull request:
Reworked web interface - mostly complete, but still work in progress. Deployed in [staging](https://apps.stg.fedoraproject.org/koschei/).

Changes include:
- new theme based on [Bootstrap 4 Alpha](https://v4-alpha.getbootstrap.com/) and [fedora-bootstrap](https://pagure.io/fedora-bootstrap/)
- no table-based layouts - Bootstrap grid is used instead
- reworked layout for some views
- icons with [Font Awesome](http://fontawesome.io/)
- different classes of flash flashes
- added human-readable time and number representation
- new logo

msimacek commented:
How is it supposed to be installed for development and production? There are absolute paths to your filesystem, so I'm not sure how I should test it

mizdebsk commented:
These are not absolute paths, but URLs relative to deployment URL: https://apps.fedoraproject.org/
I've also uploaded tarball with these resources at https://mizdebsk.fedorapeople.org/tmp/koschei-fb-global.tar.bz2

msimacek commented:
Thanks. And what is the plan for production? Would the stylesheets be packaged as RPMs?

mizdebsk commented:
For production in Fedora infrastructure, we would use resources already available on apps.fedoraproject.org, like https://apps.stg.fedoraproject.org/global/fedora-bootstrap-1.0.3/fedora-bootstrap.css
For non-Fedora-infra deployments, I don't know yet.

msimacek commented:
Would it work with unbranded bootstrap?

mizdebsk commented:
Yes.

msimacek commented:
Then what about putting the URL set in configuration and leaving the decision for the admin? Fedora deployments would use apps.fedoraproject, non-fedora deployments could use some of the public CDNs or host their own copy

mizdebsk commented:
Sure. I'll do that.

msimacek commented on static/koschei.css:
```diff
@@ -24,183 +24,6 @@
 
-html {
-    min-width: 800px;
-}
```
Can you please add the following, so that the font sizing matches other Fedora apps that use bootstrap-fedora?
````css
html {
    font-size: 14px;
}
````

msimacek commented on templates/group-detail.html:
```diff
@@ -1,20 +1,30 @@
+  <dd class="col-sm-9">
+    <ul class="list-inline">
+      {% for owner in group.owners %}
+      <li class="list-inline-item">{{ owner.name }}</li>
```
Can you please join them with commas? Also the spacing looks a bit odd, maybe it would look better without columns

mizdebsk replied:
Done.

msimacek commented on templates/package-detail.html:
```diff
@@ -1,55 +1,35 @@
-        </li>
-        {% endfor %}
-    </ul>
-</div>
```
The collection chooser was quite ugly, but it was there for a reason. When you click a package link in a multicollection view (like group packages or user packages), you always get package detail for the first collection (in fedora it's rawhide). So a user goes to listing of his packages, he sees that foo is broken in epel7, clicks on it and gets confused, because he sees that the builds are ok, because he is looking at rawhide, even though he wanted epel7. This gets much worse for deployments that have sparse collections (there are many collections, but packages usually exist in just one or two of them) - you click on a package link and you're almost always greeted with a message that the package doesn't exist in selected collection.

mizdebsk replied:
> When you click a package link in a multicollection view (like group packages or user packages), you always get package detail for the first collection (in fedora it's rawhide).

The same happens in current master too (without this PR).

Do we want to fix it? I.e not show any builds unless explicit collection is choosen.

msimacek replied:
> The same happens in current master too (without this PR).

That's true. I'm wondering when it regressed.

> Do we want to fix it? I.e not show any builds unless explicit collection is choosen.

It was quite important in deployments that didn't have a "master" collection like Fedora has. Since we currently don't maintain such deployments, I think we can keep the current behavior until we need more flexibility again.

msimacek commented on templates/package-detail.html:
```diff
@@ -1,55 +1,35 @@
-        </li>
-        {% endfor %}
-    </ul>
-</div>
```
And with the current tabs, if you didn't select a collection, none of them is displayed as active, so you don't know which collection you're looking at.

mizdebsk commented on koschei/frontend/views.py:
```diff
@@ -62,26 +63,11 @@ def proc_order(order):
-    if encoded:
-        return '?' + encoded
-    return ''
-
```
This is not just cosmetic, but there is technical reason - URL "?" is valid and means "current page with empty query args". Alternative would be specifying full relative URL. But if you insist I can work on that.

mizdebsk commented on templates/macros.html:
```diff
@@ -1,200 +1,304 @@
+    <div class="float-right">
+      <a class="btn btn-secondary btn-sm" href="{{ url_for('build_detail', build_id=build.id) }}"
+         data-toggle="tooltip" data-placement="right"
+         title="Show details of this build">...</a>
```
I've changed it to fa-info, but for me ellipsis was better.

mizdebsk commented on templates/package-detail.html:
```diff
@@ -1,55 +1,35 @@
+{% if items|length > max %}
+<div class="row">
+  <div class="col-sm-2 offset-5">
+    <a class="btn btn-secondary btn-sm kk-toggle-button"
```
I've added "mt-2" margin.

mizdebsk commented on templates/package-detail.html:
```diff
@@ -58,225 +38,275 @@
+            </div>
+
+            {% for (name, url) in generate_links(package)  %}
+            <a class="card-link" href="{{ url }}">{{ name }}</a>
```
I've made it into 2 columns as suggested.

mizdebsk commented:
@msimacek This should be reviewable. I think I've addressed all your concerns or commented on them.

> Would it work with unbranded bootstrap?

I've used some components specific to fedora-bootstrap (masthead, subheader), so this won't work with pure bootstrap any longer.

mizdebsk commented:
Thanks!
